src/pages/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,280 @@
+
+### Removed: `legacy.astroFlavoredMarkdown`
+
+In 1.0 Astro moved the old Astro Flavored Markdown (also known as Components in Markdown) to a legacy feature. This change removes the `legacy.astroFlavoredMarkdown` option completely.
+
+In 2.0 this feature is not available in Astro at all. We recommend [migrating to MDX with the integration](/en/guides/integrations-guide/mdx/) for those were still using this feature in 1.x. See our older migration guide for more about [converting existing Mardown files to MDX](https://github.com/withastro/docs/blob/aa7e4f568f418657eba297778729a8aee2a36409/src/pages/en/migrate.mdx#converting-existing-md-files-to-mdx).
+
+### Removed: `Astro.resolve()`
+
+In 0.24 Astro deprecated `Astro.resolve()`. This change removes the option completely.
+
+In 2.0 this feature is not available in Astro at all. You can resolve asset paths using `import` instead. For example:
+
+```astro
+---
+import 'style.css';
+import imageUrl from './image.png';
+---
+
+<img src={imageUrl} />
+```
+
+See the [v0.24 migration guide](https://github.com/withastro/docs/blob/aa7e4f568f418657eba297778729a8aee2a36409/src/pages/en/migrate.mdx#deprecated-astroresolve) for more information.
+
+### Removed: `Astro.fetchContent()`
+
+In 0.26 Astro deprecated `Astro.fetchContent()`. This change removes the option entirely.
+
+In 2.0 this feature is not available in Astro at all. Use [`Astro.glob()`](/en/guides/imports/#astroglob) instead to fetch Markdown files, or migrate to the [Content Collections](/en/guides/content-collections/) feature.
+
+```js
+let allPosts = await Astro.glob('./posts/*.md');
+```
+
+### Removed: `Astro.canonicalURL`
+
+In 1.0 Astro deprecated `Astro.canonicalURL`. This change removes the option entirely.
+
+In 2.0 this feature is not available in Astro at all. Use `Astro.url` instead to construct the canonical URL.
+
+```js
+const canonicalURL = new URL(Astro.url.pathname, Astro.site);
+```
+
+### Added: `_astro/` folder for build assets
+
+Astro 2.0 moves the location of all build output assets to a new `_astro/` folder. Previously, these assets were built to various locations, including `assets/`, `chunks/` and the root of the build output. 
+
+You can control this location with the [new `build.assets` configuration option](/en/reference/configuration-reference/#buildassets).
+
+If your deployment relies on the location of these assets, you may need to update your caching configuration with your deployment platform.
+
+### Changed: Markdown plugin configuration
+
+#### Removed: `extendDefaultPlugins`
+
+Astro 2.0 removes `markdown.extendDefaultPlugins`. Applying remark and rehype plugins **no longer disables** Astro's default plugins, so this option is no longer needed. 
+
+GitHub-Flavored Markdown and Smartypants are now applied whether or not custom `remarkPlugins` or `rehypePlugins` are configured. (You can opt-out manually by setting [`gfm` and `smartypants`](#added-gfm-and-smartypants) to `false` in your config.)
```

Comment:
I had trouble understanding this paragraph. If I'm a reader, how do I know if this impacts me / if I need to change something? An explicit: "if X, no action should be needed." or "if X, you'll need to Y".

Reply:
I think this is taken care of in the rewrite! At least, we can rediscuss it there.